Library/Homebrew/global.rb

```diff
@@ -26,7 +26,7 @@
 end
 RUBY_BIN = RUBY_PATH.dirname
 
-HOMEBREW_USER_AGENT = "Homebrew #{HOMEBREW_VERSION} (Ruby #{RUBY_VERSION}-#{RUBY_PATCHLEVEL}; #{OS_VERSION})"
+HOMEBREW_USER_AGENT = "#{ENV["HOMEBREW_USER_AGENT"]} (Ruby #{RUBY_VERSION}-#{RUBY_PATCHLEVEL}; #{OS_VERSION})"
```

Comment:
On second thought, that's pretty unlikely to be relevant to behavior that makes it out to servers, so probably not worth including. Doing `(Macintosh; Intel Mac OS X 10.9.0)` and maybe adding curl like Mike suggests sounds good.